Datastore/src/EntityMapper.php

```diff
@@ -211,21 +235,26 @@ public function convertValue($type, $value)
                     ? $value['properties']
                     : [];
 
-                $decoded = $this->responseToEntityProperties($properties);
+                $decoded = $this->responseToEntityProperties($properties, $className);
                 $props = $decoded['properties'];
                 $excludes = $decoded['excludes'];
 
-                if (isset($value['key'])) {
-                    $namespaceId = (isset($value['key']['partitionId']['namespaceId']))
-                        ? $value['key']['partitionId']['namespaceId']
-                        : null;
+                // @todo (major) all embedded entities should be of type Entity or given EntityInterface type.
```

Comment:
Is this todo something we should touch on before merge?

Reply:
No, it's a breaking change. It will mean discontinuing any representation of an embedded entity as an array.

Reply:
Makes sense! I read major as "this is gonna be a lot of work!" for some reason :P